packages/block-library/src/breadcrumbs/index.php

```diff
@@ -33,8 +32,23 @@ function render_block_core_breadcrumbs( $attributes, $content, $block ) {
 		);
 	}
 
-	// Handle archive pages (taxonomy, post type, date, author archives).
-	if ( is_archive() ) {
+	// Handle search results.
+	if ( is_search() ) {
+		$search_query       = get_search_query();
+		$breadcrumb_items[] = sprintf(
+			'<span aria-current="page">%s</span>',
+			/* translators: %s: search query */
+			sprintf( esc_html__( 'Search results for: "%s"' ), esc_html( $search_query ) )
+		);
+		$breadcrumb_items = block_core_breadcrumbs_maybe_add_paged( $breadcrumb_items );
+	} elseif ( is_404() ) {
+		// Handle 404 pages.
+		$breadcrumb_items[] = sprintf(
+			'<span aria-current="page">%s</span>',
+			esc_html__( '404 Not Found' )
```

Comment:
"404 Not Found" is a bit technical/geeky. Do we even need 404 to be in the breadcrumb item title? Maybe just "Page Not Found" (which would match the page title of the 404 page) or just "Not Found"?